etna/auto/auto.py

```diff
@@ -361,26 +478,108 @@ def _objective(trial: Trial) -> float:
 
         return _objective
 
-    def _init_optuna(self):
+    def _init_pool_optuna(self, suppress_logging: bool = False) -> Optuna:
         """Initialize optuna."""
-        if isinstance(self.pool, Pool):
-            pool: List[Pipeline] = self.pool.value.generate(horizon=self.horizon)
-        else:
-            pool = self.pool
+        pool = [pipeline.to_dict() for pipeline in self._pool]
+        logging_verbosity = optuna.logging.get_verbosity()
+        try:
+            if suppress_logging:
+                optuna.logging.set_verbosity(optuna.logging.ERROR)
+            pool_optuna = Optuna(
+                direction="maximize" if self.target_metric.greater_is_better else "minimize",
+                study_name=self._pool_folder,
+                storage=self.storage,
+                sampler=ConfigSampler(configs=pool),
+            )
+        finally:
+            optuna.logging.set_verbosity(logging_verbosity)
+        return pool_optuna
+
+    def _init_tuners(self, pool_optuna: Optuna) -> List["Tune"]:
+        trials = pool_optuna.study.get_trials()
+        configs = [trial.user_attrs["pipeline"] for trial in trials]
+
+        results = []
+        for config in configs:
+            cur_pipeline = get_from_params(**config)
+            cur_folder = self._get_tune_folder(cur_pipeline)
+            tuner = Tune(
+                pipeline=cur_pipeline,
+                target_metric=self.target_metric,
+                horizon=self.horizon,
+                metric_aggregation=self.metric_aggregation,
+                backtest_params=self.backtest_params,
+                experiment_folder=cur_folder,
+                runner=self.runner,
+                storage=self.storage,
+                metrics=self.metrics,
+                sampler=None,
+            )
+            results.append(tuner)
 
-        pool_ = [pipeline.to_dict() for pipeline in pool]
+        return results
 
-        optuna = Optuna(
-            direction="maximize" if self.target_metric.greater_is_better else "minimize",
-            study_name=self.experiment_folder,
-            storage=self.storage,
-            sampler=ConfigSampler(configs=pool_),
-        )
-        return optuna
+    def _make_pool_summary(self, trials: List[FrozenTrial]) -> List[dict]:
+        """Get information from trial summary."""
+        study_params = []
+        for trial in trials:
+            trial_pipeline: Optional[BasePipeline] = None
+            if "pipeline" in trial.user_attrs:
```

Comment:
Why do we need to check this?
In which cases do we have no `"pipeline"`?

Reply:
If trial has failed, e.g. some runner was stopped manually, it won't have `pipeline` attribute and we can't make `get_from_params(**trial.user_attrs.get("pipeline")` because `trial.user_attrs.get("pipeline")` returns `None` and `get_from_params` can't create an object.

Reply:
If the main reason is state of pipeline - may be we should filter out but this field?

Reply:
Ok, let's leave as is